File: frontend/src/app/core/domain/usecases/create-player.usecase.ts
import { Injectable } from '@angular/core';
import { Observable, throwError } from 'rxjs';
import { Player } from '../entities/player.entity';
import { PlayerRepository } from '../repositories/player.repository';

@Injectable({
  providedIn: 'root'
})
export class CreatePlayerUseCase {
  constructor(private playerRepository: PlayerRepository) {}

  execute(player: Player): Observable<Player> {
    // Validações de negócio/domínio adicionais
    if (!player.nomeCompleto || player.nomeCompleto.trim() === '') {
      return throwError(() => new Error('O nome completo é obrigatório.'));
    }
    if (!player.dataNascimento) {
      return throwError(() => new Error('A data de nascimento é obrigatória.'));
    }
    if (player.altura <= 0) {
      return throwError(() => new Error('A altura deve ser maior que 0 cm.'));
    }
    if (player.peso <= 0) {
      return throwError(() => new Error('O peso deve ser maior que 0 kg.'));
    }
    if (!player.posicaoPrincipal || player.posicaoPrincipal < 1 || player.posicaoPrincipal > 8) {
      return throwError(() => new Error('A posição principal é obrigatória e deve ser válida.'));
    }
    return this.playerRepository.create(player);
  }
}

File: mobile/lib/features/players/domain/usecases/create_player_usecase.dart
import '../entities/player.dart';
import '../repositories/player_repository.dart';

class CreatePlayerUseCase {
  final PlayerRepository repository;

  CreatePlayerUseCase(this.repository);

  Future<Player> call(Player player) async {
    // Validações de negócio/domínio
    if (player.nomeCompleto.trim().isEmpty) {
      throw Exception('O nome completo é obrigatório.');
    }
    if (player.altura <= 0) {
      throw Exception('A altura deve ser maior que 0 cm.');
    }
    if (player.peso <= 0) {
      throw Exception('O peso deve ser maior que 0 kg.');
    }
    if (player.posicaoPrincipal < 1 || player.posicaoPrincipal > 8) {
      throw Exception('A posição principal é obrigatória e deve ser válida.');
    }
    return await repository.create(player);
  }
}
